include/ginkgo/core/matrix/csr.hpp

```diff
@@ -38,6 +38,7 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <ginkgo/core/base/index_set.hpp>
 #include <ginkgo/core/base/lin_op.hpp>
 #include <ginkgo/core/base/math.hpp>
+#include <ginkgo/core/matrix/permutation.hpp>
```

Comment:
we also don't pull in the other matrix formats despite being convertible to them.

Reply:
I think long-term, also in light of compilation time, it might be preferable to forward-declare as much as possible in headers. But for now, I'll include it.